#!/usr/bin/env python
PACKAGE = "teleop_joystick_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("climber_setpoint_rate",             double_t, 0, "Rate to change climber setpoint (currently radians/second of the winch)",                                   0, 0, 7)
gen.add("climber_low_bound",             double_t, 0, "Lowest allowed climber setpoint (currently radians/second of the winch)",                                   0, 0, 10)
gen.add("climber_high_bound",             double_t, 0, "Highest allowed climber setpoint (currently radians/second of the winch)",                                   0, 0, 1000)
gen.add("shooter_setpoint_rate",             double_t, 0, "Rate to change shooter velocity setpoint",                                   0, 0, 20)
gen.add("left_stick_trigger_point",             double_t, 0, "Point to which the left joystick must be pushed to toggle a boolean state",                                   0, 0, 1)
gen.add("turret_setpoint_rate",             double_t, 0, "Rate to change turret setpoint (radians/second)",                                   0, 0, 1)
gen.add("turret_angle_limit",             double_t, 0, "Upper and lower bound to turret setpoint (radians)",                                   0, 0, 3.14)
gen.add("intake_setpoint_rate",             double_t, 0, "Rate to change intake percent out (percent/second)",                                   0, 0, 1)
gen.add("indexer_setpoint_rate",             double_t, 0, "Rate to change indexer velocity setpoint",                                   0, 0, 20)
gen.add("control_panel_increment",             double_t, 0, "Number of rotations to increment the control panel",                                   0, 0, 1)

exit(gen.generate(PACKAGE, "teleop_joystick_control", "TeleopJoystickCompDiagnostics"))
